LV_analysis/1_get_multiplier_LV_coverage_Pa.py

```diff
@@ -0,0 +1,335 @@
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:generic_expression] *
+#     language: python
+#     name: conda-env-generic_expression-py
+# ---
+
+# # Coverage of MultiPLIER LV
+#
+# The goal of this notebook is to examine why genes were found to be generic. Specifically, this notebook is trying to answer the question: Are generic genes found in more multiplier latent variables compared to specific genes?
+#
+# The PLIER model performs a matrix factorization of gene expression data to get two matrices: loadings (Z) and latent matrix (B). The loadings (Z) are constrained to aligned with curated pathways and gene sets specified by prior knowledge [Figure 1B of Taroni et. al.](https://www.cell.com/cell-systems/pdfExtended/S2405-4712(19)30119-X). This ensure that some but not all latent variables capture known biology. The way PLIER does this is by applying a penalty such that the individual latent variables represent a few gene sets in order to make the latent variables more interpretable. Ideally there would be one latent variable associated with one gene set unambiguously.
+#
+# While the PLIER model was trained on specific datasets, MultiPLIER extended this approach to all of recount2, where the latent variables should correspond to specific pathways or gene sets of interest. Therefore, we will look at the coverage of generic genes versus other genes across these MultiPLIER latent variables, which represent biological patterns.
+#
+# **Definitions:**
+# * Generic genes: Are genes that are consistently differentially expressed across multiple simulated experiments.
+#
+# * Other genes: These are all other non-generic genes. These genes include those that are not consistently differentially expressed across simulated experiments - i.e. the genes are specifically changed in an experiment. It could also indicate genes that are consistently unchanged (i.e. housekeeping genes)
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+
+import os
+import random
+import textwrap
+import scipy
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from sklearn.preprocessing import MinMaxScaler
+
+import rpy2.robjects as ro
+from rpy2.robjects import pandas2ri
+from rpy2.robjects.conversion import localconverter
+
+from ponyo import utils
+from generic_expression_patterns_modules import lv
+
+# +
+# Get data directory containing gene summary data
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+data_dir = os.path.join(base_dir, "pseudomonas_analysis")
+
+# Read in config variables
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "configs", "config_pseudomonas_33245.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+local_dir = params["local_dir"]
+
+project_id = params["project_id"]
+quantile_threshold = 0.97
```

Comment:
Looks like this parameter is slightly different than the human script (0.97 vs. 0.98) - is that intentional? Same question for the percentile used to define generic genes (60 in human script vs. 80 in this script).

Reply:
Thanks for checking. This was intentional. I was trying to vary the quantile to get similar numbers of high weight genes between Pa and human in case that contributed a difference in trend. The generic threshold is based on the correlation plots and so this varied based on the dataset too.